### PR TITLE
set x_RESOURCE_TOKEN_REFS to V0009

### PR DESCRIPTION
**Pull request by lmsurpre**
We missed this as part of #1996, so anyone that has applied these updates to a DB since that change now has the wrong
version of that object.  Since that change was never released, I think we're OK just to fix it now (and help anyone
riding the bleeding edge to manually fix their VERSION_HISTORY table).

I also made an unrelated tweak to the Migration interface:  priorVersion should never be null and this makes that more clear

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/Migration.java [hunk 1]
```diff
@@ -24,5 +24,5 @@
      * @implSpec this method should only be executed when the current version > priorVersion and so the steps need not be
      *         idempotent
      */
-    public List<IDatabaseStatement> migrateFrom(Integer priorVersion);
+    public List<IDatabaseStatement> migrateFrom(int priorVersion);
```
Comment (punktilious) [inline fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/Migration.java:27]: copyright

Comment (lmsurpre) [reply fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/Migration.java:27]: fixed

**Review by punktilious — APPROVED** [review]
LGTM

**Comment by punktilious** [comment]
If an attempt to apply a schema update fails with errors like the following:

```
2021-03-04 14:35:07.959 0000000d  SEVERE .fhir.task.core.impl.TaskGroup GROUP:FHIRDATA.RELATEDPERSON_RESOURCE_TABLE_GROUP:0

org.postgresql.util.PSQLException: ERROR: constraint "pk_relatedperson_resource_token_refs" of relation "relatedperson_resource_token_refs" does not exist
com.ibm.fhir.database.utils.api.UndefinedNameException: org.postgresql.util.PSQLException: ERROR: constraint "pk_relatedperson_resource_token_refs" of relation "relatedperson_resource_token_refs" does not exist
	at com.ibm.fhir.database.utils.postgres.PostgresTranslator.translate(PostgresTranslator.java:100)
	at com.ibm.fhir.database.utils.common.DropPrimaryKey.run(DropPrimaryKey.java:54)
	at com.ibm.fhir.database.utils.common.CommonDatabaseAdapter.runStatement(CommonDatabaseAdapter.java:459)
	at com.ibm.fhir.database.utils.postgres.PostgresAdapter.runStatement(PostgresAdapter.java:266)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)
	at java.base/java.util.stream.ReferencePipeline$Head.forEachOrdered(ReferencePipeline.java:668)
	at com.ibm.fhir.database.utils.model.Table.apply(Table.java:148)
	at com.ibm.fhir.database.utils.model.BaseObject.applyVersion(BaseObject.java:263)
	at com.ibm.fhir.database.utils.model.ObjectGroup.applyVersion(ObjectGroup.java:67)
	at com.ibm.fhir.database.utils.model.BaseObject.applyTx(BaseObject.java:214)
	at com.ibm.fhir.database.utils.model.BaseObject.lambda$collect$0(BaseObject.java:201)
	at com.ibm.fhir.task.core.impl.TaskGroup.runTask(TaskGroup.java:137)
	at com.ibm.fhir.task.core.impl.TaskGroup.lambda$runTask$0(TaskGroup.java:129)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: org.postgresql.util.PSQLException: ERROR: constraint "pk_relatedperson_resource_token_refs" of relation "relatedperson_resource_token_refs" does not exist
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2553)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2285)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:323)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:473)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:393)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:322)
	at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:308)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:284)
	at org.postgresql.jdbc.PgStatement.executeUpdate(PgStatement.java:258)
	at com.ibm.fhir.database.utils.common.DropPrimaryKey.run(DropPrimaryKey.java:52)
	... 16 more
```

then the version_history table can be repaired by executing the following statement against the database:

```
update fhir_admin.version_history set version = 9 where object_type = 'TABLE' and object_name like '%_RESOURCE_TOKEN_REFS' and version = 8;
commit;
```

The error and fix for Db2 or Derby would be similar.

**Pull request merged**